File: tools/cpp/tests/statistics/summary/test_WindowedFame.cc
// unittest_windowed_fame

#include <gtest/gtest.h>
#include "WindowedFAME.h"
#include <vector>

using KML::Statistics::WindowedFAME;

class WindowedFAMETest : public ::testing::Test
{
    public:
        WindowedFAME* fm;

        WindowedFAMETest()
        {
            fm = new WindowedFAME(0.1, 0.3);
        }
};

TEST_F(WindowedFAMETest, TestEmptyEvaluate)
{
    EXPECT_FLOAT_EQ(0.0, fm->evaluate());
}

TEST_F(WindowedFAMETest, TestMedianConstant)
{
    std::vector<double> data{10, 10, 10};
    fm->update(data);
    EXPECT_FLOAT_EQ(10.0, fm->evaluate());
}

File: tools/python/tests/statistics/summary/test_WindowedFame.py
import unittest

from KML.statistics.summary.WindowedFAME import PyWindowedFAME


class TestWindowedFAME(unittest.TestCase):

    WINDOW_SIZE = 5
    DATA = [10.0] * WINDOW_SIZE
    FM = None

    def setUp(self):
        self.FM = PyWindowedFAME()

    def tearDown(self):
        self.FM = None

    def test_update(self):
        self.FM.update(self.DATA)

    def test_constant_median(self):
        self.FM.update(self.DATA)
        self.assertEqual(self.FM.evaluate(), 10.0)
